controllers/events.js

```diff
@@ -0,0 +1,63 @@
+const eventRouter = require('express').Router()
+const verifyService = require('../services/verifyService')
+const eventService = require('../services/eventService')
+
+// TODO: verify if the scout is actually logged in ..?
+
+
+// Get a list of scouts events
+eventRouter.get('', async (req, res) => {
+  const scout = req.session.scout
+  if (!scout){
+    res.status(403).send('you are not logged in!')
+  }else{
+    const events = await eventService.getAllEvents(scout.id)
+    res.status(200).send(events) 
+  }
+})
+
+// Add a new event, return the added event
+eventRouter.post('', async (req, res) => {
+  const scout = req.session.scout
+  if (!scout){
+    res.status(403).send('You are not logged in!')
+  }else{
+    const event = await eventService.createEvent(scout.id, req.body)
+    res.status(200).send(event)
+  }
+})
+
+// Edit an event, return the edited event
+eventRouter.put('/:eventId', async (req, res) => {
+  const scout = req.session.scout
+  const eventId = req.params.eventId
+  if (!verifyService.scoutOwnsEvent(scout, eventId)){
+    res.status(403).send('You are not the owner of this event!')
+  }else{
+    const event=await eventService.updateEvent(eventId, req.body)
+    if (event.error){ //Should never really happen since verifyService should prevent all errors
+      res.status(404).send(event.error)
+    }else{
+      res.status(200).send(event)
+    }
+  }
+})
+
+// Delete an event
+eventRouter.delete('/:eventId', async (req, res) => {
+  const scout = req.session.scout
+  const eventId = req.params.eventId
```

Comment:
Should this use `parseInt(req.params.eventId)`? At least weird calls like /events/ghfdgr should be handled somehow. Same goes for /activities.

Reply:
Should we do the validations in the controllers or in the services?

Reply:
I think controllers should deal with that, services shouldn't need to worry about it.

Reply:
Adding parseInt isn't enough, because it returns nan when no integer could be parsed, and Sequelize doesn't like nan;s it when it's expecting an integer.

Reply:
Ok, so return 400 or something for nan?